### ISSUE TITLE
`vcpkg search` freezes when `vcpkg.json` is present in the folder

### ISSUE THREAD

**Issue by PoetaKodu** [microsoft/vcpkg#31133]
**Describe the bug**

In the title. When I remove `vcpkg.json` it works again.

**Environment**
- OS: Windows 11 x64
- Compiler: Latest MSVC from Visual Studio 2022 Preview (irrelevant)

**To Reproduce**
1. Fresh install of the vcpkg (clone the git repo)
2. Bootstrap the vcpkg
3. Add `C:\dev\vcpkg` to PATH
4. Create a folder "myproject"
5. Create a `vcpkg.json` file within it with the following content `{}` (empty JSON object)
6. Run `vcpkg x-update-baseline --add-initial-baseline`
  In my case it wrote the following baseline: `70992f64912b9ab0e60e915ab7421faa197524b7`
7. Run `vcpkg search fmt`
8. The application runs indefinitely.

**Expected behavior**
Shouldn't freeze.

**Failure logs**
- None

**Additional context**

I recorded a video:
https://user-images.githubusercontent.com/6839845/234612008-a4b1d1af-3a95-45af-83ca-8269fe61854d.mp4

**Comment by PoetaKodu** [comment]
Ah... now I see that it probably should've been reported in `vcpkg-tool` repo

**Comment by jimwang118** [comment]
According to your reproduction steps, I deleted the `vcpkg.json` file after executing `vcpkg search fmt` locally, and then executed `vcpkg search fmt` again, but the `vcpkg.json` file was not restored. I don't quite understand what you mean. After executing the `vcpkg search fmt` command, the `vcpkg.json` file will be restored, or the execution results of the two commands are the same?

**Comment by PoetaKodu** [comment]
The problem is that as long as there is a `vcpkg.json` file with default-generated baseline field I cannot run `vcpkg search fmt` (and possibly other) commands. The package manager clearly shouldn't run indefinitely whatever is present in a `vcpkg.json` file.

**Comment by BillyONeal** [comment]
OK, it looks like it isn't actually hung, but it's trying to do individual git checkouts on each possible port so that the description can be searched, which is horribly slow. Maybe we should search only by name in these cases...

**Comment by BillyONeal** [comment]
OK in my testing on a 12900HX it does finish, but takes about 4.5 minutes to do so.

**Comment by jimwang118** [comment]
No stale

**Comment by jimwang118** [comment]
This issue has been fixed by PR [1120](https://github.com/microsoft/vcpkg-tool/pull/1120). Local test passed.

**Comment by BillyONeal** [comment]
> This issue has been fixed by PR [1120](https://github.com/microsoft/vcpkg-tool/pull/1120). Local test passed.

I don't think this is entirely fixed by that... that helped, but we really need the `mktree` part to truly make git registries perform like the builtin registry.

**Comment by BillyONeal** [comment]
Not stale :(

**Comment by explorer-exe-bilibili** [comment]
我也遇到了相同的问题，只要目录下存在vcpkg.json就无法使用search，会一直卡在那里，CPU占用1%左右，磁盘0.1MB/s，不清楚vcpkg的操作，版本是2025-04-16-f9b6c6917b23c1ccf16c1a9f015ebabf8f615045

**Comment by explorer-exe-bilibili** [comment]
I also encountered the same problem, as long as there is a vcpkg.json in the directory, I can't use search, it will always be stuck there, the CPU occupies about 1%, the disk is 0.1MB/s, I don't know the operation of vcpkg, the version is 2025-04-16-f9b6c6917b23c1ccf16c1a9f015ebabf8f615045

**Comment by BillyONeal** [comment]
@explorer-exe-bilibili Unfortunately us waiting for launching thousands of child processes doesn't show up in task manager easily because a lot of the time is not in our process :(
